java/src/main/java/org/rocksdb/FlinkCompactionFilter.java

```diff
@@ -0,0 +1,142 @@
+//  Copyright (c) 2017-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+
+package org.rocksdb;
+
+/**
+ * Just a Java wrapper around FlinkCompactionFilter implemented in C++.
+ *
+ * Note: this compaction filter is a special implementation, designed for usage only in Apache Flink project.
+ */
+public class FlinkCompactionFilter
+    extends AbstractCompactionFilter<Slice> {
+  public enum StateType {
+    // WARNING!!! Do not change the order of enum entries as it is important for jni translation
+    Disabled,
```

Comment:
Maybe you could add a unit tests that checks all ordinal values against constants to prevent changes in the order.

Reply:
should be in `FlinkCompactionFilterTest.checkStateTypeEnumOrder`